sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -988,7 +989,15 @@ private[hive] class HiveClientImpl(
 private[hive] object HiveClientImpl extends Logging {
   /** Converts the native StructField to Hive's FieldSchema. */
   def toHiveColumn(c: StructField): FieldSchema = {
-    val typeString = HiveVoidType.replaceVoidType(c.dataType).catalogString
+    val typeString = if (c.metadata.contains(CHAR_VARCHAR_TYPE_STRING_METADATA_KEY)) {
```

Comment:
where do we hide the char/varchar from schema before reaching here?

Reply:
HiveTableScanExec -> HiveTableRelation -> CatalogTable, we need this chage to covert it back to `HiveTable` properly